packages/candy/src/io/file.candy

```diff
@@ -112,18 +123,30 @@ class Path {
       if segment is ChildSegment {
         normalized.add(segment)
       } else if segment is ParentSegment {
-        normalized.removeLast()
+        if normalized.isEmpty() {
+          normalized.add(ParentSegment())
+        } else {
+          normalized.removeLast()
+        }
       }
     }
     Path.unsafe(normalized, isAbsolute)
   }
 
   // TODO(marcelgarus): Getterize!
   fun parent(): Option<Path> {
-    if segments.isEmpty {
+    while segments.last() is Option<CurrentSegment> {
```

Comment:
Shouldn't we make a mutable copy of that?

Reply:
Jup